odoo/tools/translate.py

```diff
@@ -1570,7 +1570,9 @@ def _get_translation_upgrade_queries(cr, field):
               GROUP BY it.res_id
             )
             UPDATE {Model._table} m
-               SET "{field.name}" = CASE WHEN t.noupdate THEN m."{field.name}" || t.value ELSE t.value || m."{field.name}" END
+               SET "{field.name}" = CASE WHEN t.noupdate IS FALSE THEN t.value || m."{field.name}"
+                                         ELSE m."{field.name}" || t.value
+                                     END
```

Comment:
So the case `t.noupdate IS NULL` is handled just like `t.noupdate IS TRUE`, right?
I don't understand this, as we usually coerce `NULL` to `FALSE`.

Reply:
Here the `noupdate is NULL` is because the record doesn't have an external id. So it is not updatable by xml(data) import or po(translation) import. The 'en_US' value for old translations is a better translation than the original source value in the column.

For example
record column `{"en_US": "old_src_value"}` (it is auto converted from the old column value "old_src_value")
grouped translations `{"en_US": "old_en_US", "fr_FR": "old_fr_FR"}` (from ir_translation)
`"old_en_US"` is better for `"old_src_value"` as the new en_US value

Reply:
Thanks for the explanation, as it is not trivial!